### PR TITLE
Tests for FDA

### LINKED ISSUE #26037: add unit tests for FDA integration
Adding unit tests for PR #26038

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
This work adds the missing tests for the FDA GUI. 

**To test:**

Fixes #26037. 

---
Does not need to be in release notes as it just adds tests
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/transform_widget_new_test.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+
+# pass the call not the object to widget
+
+@start_qapplication
+class Transform2Test(unittest.TestCase):
+    def setUp(self):
+        #self.load = mock.MagicMock()
+        self.context = setup_context(True)
+        #self.fft = FFTWidget(self.context)
```
Comment (AnthonyLim23) [inline scripts/test/Muon/transform_widget_new_test.py]:
```suggestion
```

File: scripts/test/Muon/transform_widget_new_test.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+from Muon.GUI.Common.test_helpers.context_setup import setup_context
+from Muon.GUI.Common.observer_pattern import GenericObservable
+
+
+# pass the call not the object to widget
+
+@start_qapplication
+class Transform2Test(unittest.TestCase):
+    def setUp(self):
+        #self.load = mock.MagicMock()
```
Comment (AnthonyLim23) [inline scripts/test/Muon/transform_widget_new_test.py]:
```suggestion
```

File: scripts/test/Muon/transform_widget_new_test.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        self.context = setup_context(True)
+        #self.fft = FFTWidget(self.context)
+        #self.maxent = MaxEntWidget(self.acontext)
```
Comment (AnthonyLim23) [inline scripts/test/Muon/transform_widget_new_test.py]:
```suggestion
```

File: scripts/test/Muon/muon_context_with_frequency_test.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    def populate_ADS(self):
+        self.context.calculate_all_groups()
+        self.context.show_all_groups()
+        self.context.calculate_all_pairs()
+        self.context.show_all_pairs()
+        workspace = CreateWorkspace([0], [0], StoreInADS=False)
+        self.context.phase_context.add_phase_quad(
+            MuonWorkspaceWrapper(workspace, 'EMU19489; PhaseQuad; PhaseTable EMU19489'))
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/muon_context_with_frequency_test.py]:
```suggestion
```

File: scripts/test/Muon/home_plot_widget_model_test.py [hunk 5]
```diff
@@ -36,5 +36,44 @@ def test_plot_creates_new_plot_window_and_plots_workspace_list(self, mock_plot):
...
+        mock_plot = mock.MagicMock()
+        self.model.plot_figure = mock_plot
+        workspace_list = []
+        subplot_title = 'MUSR62260 bottom'
+
+        self.assertEquals(self.model.plot(workspace_list, subplot_title, 'Time', False, 'Muon Analysis'), mock_plot)
+        self.assertEquals(self.model.plotted_workspaces, [])
+        self.assertEquals(self.model.plotted_workspaces_inverse_binning,{})
+        self.assertEquals(self.model.plotted_fit_workspaces, [])
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/home_plot_widget_model_test.py]:
```suggestion
```

File: scripts/test/Muon/home_plot_widget_freq_test.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    def test_get_workspaces_to_plot_freq(self):
+        self.view.get_selected.return_value =  "Frequency Re"
+        self.context.data_context.current_runs = [[62260]]
+        self.context.get_names_of_frequency_domain_workspaces_to_fit = mock.Mock()
+        self.presenter.get_freq_workspaces_to_plot("fwd", self.view.get_selected() )
+
+        self.assertEquals(self.context.get_names_of_frequency_domain_workspaces_to_fit.call_count, 1)
+        self.context.get_names_of_frequency_domain_workspaces_to_fit.assert_called_once_with("62260", "fwd", True, "Re")
+
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/home_plot_widget_freq_test.py]:
```suggestion
```

File: scripts/test/Muon/home_plot_widget_freq_test.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        self.assertEquals(self.context._frequency_context.plot_type, "")
+
+
+    def test_get_workspace_to_plot(self):
+        self.view.get_selected.return_value =  "Frequency Re"
+        self.presenter.get_freq_workspaces_to_plot = mock.Mock()
+
+        self.presenter.get_workspaces_to_plot("fwd",True, self.view.get_selected() )
+        self.assertEquals(self.presenter.get_freq_workspaces_to_plot.call_count, 1)
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/home_plot_widget_freq_test.py]:
```suggestion
```

File: scripts/test/Muon/home_plot_widget_freq_test.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        self.assertEquals(self.context._frequency_context.plot_type, "Re")
+
+    def test_plot_type_changed_to_time(self):
+        self.view.get_selected.return_value =  "Frequency Re"
+
+        self.view.get_selected.return_value =  "Asymmetry"
+        self.presenter.handle_plot_type_changed()
+        self.assertEquals(self.context._frequency_context.plot_type, "")
+
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/home_plot_widget_freq_test.py]:
```suggestion
```

File: scripts/test/Muon/frequency_domain_context_test.py [hunk 9]
```diff
@@ -6,46 +6,132 @@
...
+
+    def test_get_freq_names_FFT_run(self):
+        self.context.add_FFT(ws_freq_name=FFT_NAME_COMPLEX_RE, Re_run="62260",Re= "top", Im_run="62261", Im="fwd",phasequad=False)
+        self.context.add_FFT(ws_freq_name=FFT_NAME_COMPLEX_IM, Re_run="62260",Re= "top", Im_run="62261", Im="fwd",phasequad=False)
+        self.context.add_FFT(ws_freq_name=FFT_NAME_COMPLEX_MOD, Re_run="62260",Re= "top", Im_run="62261", Im="fwd",phasequad=False)
+
+        output = self.context.get_frequency_workspace_names(run_list=[["62261","62262"]],group=["fwd","top"],pair=[],phasequad=False,frequency_type="FFT All")
+        six.assertCountEqual(self,output,[FFT_NAME_COMPLEX_RE, FFT_NAME_COMPLEX_IM, FFT_NAME_COMPLEX_MOD])
+
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/frequency_domain_context_test.py]:
```suggestion
```

File: scripts/test/Muon/frequency_domain_context_test.py [hunk 10]
```diff
@@ -6,46 +6,132 @@
...
+    def test_get_freq_names_FFT_no_Im_Mod_parts(self):
+        output = self.context.get_frequency_workspace_names(run_list=[["62260"]],group=["fwd","top"],pair=[],phasequad=False,frequency_type="Mod")
+        six.assertCountEqual(self,output,[FFT_NAME_RE_MOD])
+
+    def test_get_freq_names_FFT_no_Im_Im_parts(self):
+        output = self.context.get_frequency_workspace_names(run_list=[["62260"]],group=["fwd","top"],pair=[],phasequad=False,frequency_type="Im")
+        six.assertCountEqual(self,output,[])
+
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/frequency_domain_context_test.py]:
```suggestion
```

File: scripts/test/Muon/fitting_tab_widget/workspace_selector_dialog_presenter_test.py [hunk 11]
```diff
@@ -36,6 +36,31 @@ def test_handle_group_pair_selection_changed(self):
...
+        self.context.get_names_of_workspaces_to_fit.reset_mock()
+
+    def set_combo(self, name):
+        index = self.view.time_domain_combo.findText(name)
+        self.view.time_domain_combo.setCurrentIndex(index)
+
+    def test_is_it_freq_flase(self):
+        self.set_combo("Time Domain")
+        self.assertEquals(self.view.is_it_freq, "None")
+
```
Comment (AnthonyLim23) [inline scripts/test/Muon/fitting_tab_widget/workspace_selector_dialog_presenter_test.py]:
```suggestion
```

**Review by Harrietbrown — COMMENTED** [review]
Just one small point, otherwise its good.

File: scripts/test/Muon/home_plot_widget_model_test.py [hunk 12]
```diff
@@ -36,5 +36,43 @@ def test_plot_creates_new_plot_window_and_plots_workspace_list(self, mock_plot):
...
+    @mock.patch('Muon.GUI.Common.home_plot_widget.home_plot_widget_model.plot')
+    def test_plot_logic_force_new(self,mock_plot):
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply scripts/test/Muon/home_plot_widget_model_test.py]: We would still need both parts here. The first part creates a plot and then ensures that we have. The second part then ensures that we are over writing the old plot.

**Review by Harrietbrown — APPROVED** [review]
All tests pass and is written well.

**Pull request merged**